File: c/gemini/p4_fc.c
#include <stdio.h>
#include <stdlib.h>
#include <errno.h>
#include <limits.h>

int main() {
    const char *vehicles[] = {"Car", "Bike", "Bus", "Train", "Plane"};
    int num_vehicles = 5;
    char input[256];

    printf("Enter vehicle index (0-%d): ", num_vehicles - 1);
    if (fgets(input, sizeof(input), stdin) == NULL) {
        fprintf(stderr, "Error reading input.\n");
        return 1;
    }

    char *endptr;
    errno = 0;
    long index = strtol(input, &endptr, 10);

    if (errno != 0 || input == endptr) {
        fprintf(stderr, "Invalid integer input.\n");
        return 1;
    }

    if (index < 0 || index >= num_vehicles) {
        fprintf(stderr, "Index out of range.\n");
        return 1;
    }

    printf("Vehicle: %s\n", vehicles[index]);
    return 0;
}

File: python/gemini/p4_fc.py
import sys

def main():
    vehicles = ["Car", "Bike", "Bus", "Train", "Plane"]
    
    try:
        user_input = input(f"Enter vehicle index (0-{len(vehicles)-1}): ")
        index = int(user_input)
        
        if 0 <= index < len(vehicles):
            print(f"Vehicle: {vehicles[index]}")
        else:
            print("Index out of range.", file=sys.stderr)
            sys.exit(1)
            
    except ValueError:
        print("Invalid integer input.", file=sys.stderr)
        sys.exit(1)
    except EOFError:
        print("Error reading input.", file=sys.stderr)
        sys.exit(1)

if __name__ == "__main__":
    main()
